Project.toml

mileslucas opened the pull request:
In this pull request I've changed 3(ish) things.

1. I've added a `Project.toml` to work with the new Julia standards. I've copied the requirements from `REQUIRE` and added an appropriate `[compat]` section. 
2. I've bumped the version to 0.3.0 in the `Project.toml`
3. I've switched `° Ra` to `Ra` in the *import_from_unitful* section since default Unitful has removed the degree symbol

mileslucas committed:
created project.toml and added Manifest.toml to gitignore

mileslucas committed:
remove old require file

mileslucas committed:
remove "degrees rankine" which is not importable from Unitful

mileslucas committed:
added rankine back

giordano commented:
Thanks!

> I've switched `° Ra` to `Ra` in the `import_from_unitful` section since default Unitful has removed the degree symbol

Do you know in which version of `Unitful` this change happened?  It's important because it'd be probably good to raise the compatible Unitful version.

mileslucas committed:
update compat for unitful and add 1.1 to travis matrix

mileslucas commented:
It looks like [here](https://github.com/PainterQubits/Unitful.jl/commit/09ff6b17b171c93bc9dfa57fa87dfcb7ea2a381c) is where the code was changed, which was release `v0.13.0`. I've updated the `compat` to be `≥ 0.13.0`.

I've also added julia 1.1 to travis matrix

mileslucas commented:
It looks like this causes julia 0.7 to fail at the dependency graph with Unitful and UnitfulAngles

giordano commented:
I'd drop support for Julia 0.7.  It has been released almost one year ago, it was meant only for a transition from 0.6 to 1.0, there is no much reason now for people to stay on 0.7

mileslucas committed:
remove julia 0.7 support

mileslucas committed:
fix version issues on appveyor

mileslucas commented:
I've removed 0.7 from `compat`, appveyor and the travis matrix, it should pass the CI now.

giordano commented on Project.toml:
```diff
@@ -0,0 +1,18 @@
+name = "UnitfulAstro"
+uuid = "f2a1aa4e-5634-4978-a6f6-8c9e368f14ef"
+license = "MIT"
+version = "0.3.0"
```
Contrary to whatever I may have told you in the past, this field should be equal to the latest released version, `0.2.0` currently (no strange `-DEV` around).  The version will be bumped in the commit that we'll be used for the new release.

giordano commented on .travis.yml:
```diff
@@ -4,8 +4,8 @@ os:
 julia:
-  - 0.7
   - 1.0
+  - 1.1
```
I think that also Julia 1.2 should be available for CI now

giordano commented on appveyor.yml:
```diff
@@ -1,7 +1,7 @@
   matrix:
-  - julia_version: 0.7
   - julia_version: 1
+  - julia_version: 1.1
```
The same as in `.travis.yml`: Julia 1.2 should be available for CI

giordano reviewed (changes requested):
Sorry, I forgot to make my final comments the other day!  Thanks for the PR, I only have a few minor requests, after that we're good to go!

Commit:
update version number and add julia 1.2 to CI

mileslucas reviewed (commented):
Updated with latest commit!

giordano commented:
Thanks!

giordano commented on Project.toml:
```diff
@@ -0,0 +1,18 @@
+name = "UnitfulAstro"
+uuid = "f2a1aa4e-5634-4978-a6f6-8c9e368f14ef"
```
How did you get this UUID?  It's wrong

mileslucas replied:
Ah, I had just randomly generated a UUID, I did not think to check for the UUID from the julia registry

giordano replied:
```julia
julia> using Pkg

julia> Pkg.METADATA_compatible_uuid("UnitfulAstro")
UUID("6112ee07-acf9-5e0f-b108-d242c714bf9f")
```
:wink:
